### PR TITLE
[WIP]

### PR DESCRIPTION
**Pull request by HernanRivasAcosta**

### CODE REVIEW

File: src/sr_test_utils.erl.old [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  % Create a new element
+  [RequestBody] = maps:get(parameters, Post),
+  PostSchema = maps:get(schema, RequestBody),
+  % For now, we are only adding the required fields
+  Definitions = maps:get(<<"properties">>, PostSchema),
+  RequiredFields = maps:get(<<"required">>, PostSchema),
+  Json = [build_field(F, Definitions) || F <- RequiredFields],
+
+  io:format("Json: ~p~n", [Json]),
+  R = api_call(Post, Path, [{<<"Content-Type">>, ContentType}], maps:from_list(Json)),
```
Comment (elvisci) [inline src/sr_test_utils.erl.old:31]: Line 31 is too long:   R = api_call(Post, Path, [{<<"Content-Type">>, ContentType}], maps:from_list(Json)),.

File: src/example_SUITE.erl [hunk 2]
```diff
@@ -0,0 +1,11 @@
+-module(example_SUITE).
+
+-mixin([{sr_test_SUITE, [method_not_allowed/1]}]).
+
+-exports([init_per_suite/1]).
+
+init_per_suite(Config) ->
```
Comment (gadgetci) [inline src/example_SUITE.erl:7]: According to **Compiler**:

> Warning: function init_per_suite/1 is unused

File: src/sr_test_SUITE.erl [hunk 3]
```diff
@@ -0,0 +1,14 @@
+-module(sr_test_SUITE).
+
+-spec method_not_allowed(any()) -> ok.
+method_not_allowed(Config) ->
```
Comment (gadgetci) [inline src/sr_test_SUITE.erl:4]: According to **Compiler**:

> Warning: function method_not_allowed/1 is unused

File: src/sr_test_utils.erl [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  ok.
+
+-spec advanced_tests([module()]) -> ok.
+advanced_tests(Handlers) ->
+  ok.
+
+api_call(Method, Path) ->
+  api_call(Method, Path, []).
+
+api_call(Method, Path, Hdrs) when is_list(Hdrs) ->
```
Comment (gadgetci) [inline src/sr_test_utils.erl:26]: According to **Compiler**:

> Warning: missing specification for function api_call/3

File: src/sr_test_utils.erl [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+       Handler <- Handlers],
+
+  % Wrong content types
+  ok.
+
+-spec advanced_tests([module()]) -> ok.
+advanced_tests(Handlers) ->
+  ok.
+
+api_call(Method, Path) ->
```
Comment (gadgetci) [inline src/sr_test_utils.erl:23]: According to **Compiler**:

> Warning: missing specification for function api_call/2

File: src/sr_test_utils.erl [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  % Clean up all the repos
+  ct:comment("Clean up all the repos"),
+  _ = [sumo:delete_all(sumo_internal:schema_name(Handler:sumo_schema())) ||
+       Handler <- Handlers],
+
+  % Wrong content types
+  ok.
+
+-spec advanced_tests([module()]) -> ok.
+advanced_tests(Handlers) ->
```
Comment (gadgetci) [inline src/sr_test_utils.erl:20]: According to **Compiler**:

> Warning: variable 'Handlers' is unused

File: src/sr_test_utils.erl [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  ok.
+
+api_call(Method, Path) ->
+  api_call(Method, Path, []).
+
+api_call(Method, Path, Hdrs) when is_list(Hdrs) ->
+  api_call(Method, Path, Hdrs, <<>>).
+
+api_call(Method, Path, Hdrs, Json) when is_map(Json) ->
```
Comment (gadgetci) [inline src/sr_test_utils.erl:29]: According to **Compiler**:

> Warning: missing specification for function api_call/4

File: src/example_SUITE.erl [hunk 8]
```diff
@@ -0,0 +1,11 @@
+-module(example_SUITE).
+
+-mixin([{sr_test_SUITE, [method_not_allowed/1]}]).
+
+-exports([init_per_suite/1]).
```
Comment (elbrujohalcon) [inline src/example_SUITE.erl:5]: it's `export`

File: src/sr_test_utils.erl [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+  api_call(Method, Path, Hdrs, jsx:encode(Json));
+api_call(Method, Path, Headers, Body) ->
+  {ok, Application} = application:get_application(),
+  Port = integer_to_binary(application:get_env(Application, http_port, 8080)),
```
Comment (elbrujohalcon) [inline src/sr_test_utils.erl:33]: Request the port as a parameter (or the whole server name, maybe). Not all apps keep the port in the same place.

File: src/sr_test_utils.erl [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+      end;
+    {error, Error} ->
+      ct:fail("Couldnt ~p ~p: ~p", [Method, Path, Error])
+  catch
+    _:X ->
+      ct:log("Error: ~p; Stack: ~p", [X, erlang:get_stacktrace()]),
+      ct:fail("Couldnt ~p ~p: ~p", [Method, Path, X])
+  end.
+
+%% @todo remove once inaka/sumo_rest#44 is fixed
```
Comment (elbrujohalcon) [inline src/sr_test_utils.erl:54]: It's fixed already. Check #44

File: src/sr_test_utils.erl.old (hunk 1 above)
Comment (elvisci) [inline src/sr_test_utils.erl.old:31]: Line 31 is too long:   R = api_call(Post, Path, [{<<"Content-Type">>, ContentType}], maps:from_list(Json)),.

**Pull request closed without merging**